### DISCUSSION TITLE
How does etcd handle the issue where a client is unable to contact the cluster due to endpoint and cluster configuration mismatches after configuration changes?

### DISCUSSION THREAD

**Discussion by Phoenix500526** [etcd-io/etcd#18159] [Q&A] [answered]
During the configuration change process in etcd, it is possible for a client's endpoint to become inconsistent with the current configuration of the etcd cluster. For example, if a client has not communicated with the cluster for a long time, and during this period the cluster has undergone several configuration changes, the client will be unable to communicate with the cluster. Some companies configure an API gateway, such as APISIX, between the client and the etcd cluster. Is this the best practice to address this issue? Or, does etcd have a best practice for this situation?

**Comment by serathius** [comment, marked as answer]
https://pkg.go.dev/go.etcd.io/etcd/client/v3#Client.Sync
